uportal-war/src/main/java/org/jasig/portal/portlets/dynamicskin/FileSystemDynamicSkinService.java

```diff
@@ -249,4 +250,33 @@ public String calculateTokenForCurrentSkin(PortletRequest request) {
         return Integer.toString(hash);
     }
 
+    @Override
+    public SortedSet<String> getSkinNames(PortletRequest request) {
+        // Context to access the filesystem
+        PortletContext ctx = request.getPortletSession().getPortletContext();
+
+        // Determine the full path to the skins directory
+        String skinsFilepath = ctx.getRealPath(rootFolder);
+
+        // Create File object to access the filesystem
+        File skinsDir = new File(skinsFilepath);
```

Comment:
Sooo...

I'm not sure it matters enough to bother with now, but I wonder if we'd be better served by getting the list of skins from the skinList.xml file.  (This file normally controls which skins are selectable in the personalization-gallery UI.)

Technically Respondr skins can work just swimmingly without an entry in that file, but...
- We presumably have a routine somewhere for reading that file and figuring out what skins there are;  it would be attractive to do that one way, and perhaps even enhance it to be more powerful/flexible
- This path would give us a way to specify which skins we want to offer;  scanning the directory will not allow us to exclude defaultSkin.less as a potential choice